client_perms/server.sh

```diff
@@ -0,0 +1,278 @@
+	chmod 755 create-export-ganesha.sh dbus-send.sh
+
+	/usr/libexec/ganesha/create-export-ganesha.sh /etc/ganesha on ${GLUSTER_VOLUME}
+	/usr/libexec/ganesha/dbus-send.sh /etc/ganesha on ${GLUSTER_VOLUME}
+
+	# wait till server comes out of grace period
+	sleep 90
+
+	conf_file="/etc/ganesha/exports/export."${GLUSTER_VOLUME}".conf"
+
+	#Parsing export id from volume export conf file
+	export_id=$(grep 'Export_Id' ${conf_file} | sed 's/^[[:space:]]*Export_Id.*=[[:space:]]*\([0-9]*\).*/\1/')
+
+	# basic check if the export is available, some debugging if not
+	if ! showmount -e | grep -q -w -e "${GLUSTER_VOLUME}"
+	then
+		echo "+++ /var/log/ganesha.log +++"
+		cat /var/log/ganesha.log
+		echo
+		echo "+++ /etc/ganesha/ganesha.conf +++"
+		grep --with-filename -e '' /etc/ganesha/ganesha.conf
+		echo
+		echo "+++ /etc/ganesha/exports/*.conf +++"
+		grep --with-filename -e '' /etc/ganesha/exports/*.conf
+		echo
+		echo "Export ${GLUSTER_VOLUME} is not available"
+		exit 1
+	fi
+
+	echo "============Displaying Initial Configuration File============================="
+	cat /etc/ganesha/exports/export.${GLUSTER_VOLUME}.conf
+
+	#Enabling ACL for the volume if ENABLE_ACL param is set to True
+	if [ "${ENABLE_ACL}" == "True" ]
+	then
+	  sed -i s/'Disable_ACL = .*'/'Disable_ACL = false;'/g ${conf_file}
+	  cat ${conf_file}
+
+	  dbus-send --type=method_call --print-reply --system  --dest=org.ganesha.nfsd /org/ganesha/nfsd/ExportMgr  org.ganesha.nfsd.exportmgr.UpdateExport string:/etc/ganesha/exports/export.${GLUSTER_VOLUME}.conf string:"EXPORT(Export_Id = 2)"
+	fi
+
+fi
+
+if [ "$1" = "server_stage1" ]
+then
+	echo "======= SERVER STAGE 1 =========="
+	touch clientBlock.txt
+	echo -e "CLIENT{
+	\tClients = ${CLIENT};
+        \tSquash = \"no_root_squash\";
+        \tAccess_type = \"RO\";
+        \tProtocols = \"3\",\"4\";
+	}" > clientBlock.txt
+	
+	conf_file="/etc/ganesha/exports/export."${GLUSTER_VOLUME}".conf"
+
+	#Parsing export id from volume export conf file
+	export_id=$(grep 'Export_Id' ${conf_file} | sed 's/^[[:space:]]*Export_Id.*=[[:space:]]*\([0-9]*\).*/\1/')
+
+	echo "=======CLIENTBLOCK========="
+	cat clientBlock.txt
+	sed -i '19e cat clientBlock.txt' ${conf_file}
+
+	echo "UPDATED EXPORT FILE"
+	cat ${conf_file}
+
+	dbus-send --type=method_call --print-reply --system  --dest=org.ganesha.nfsd /org/ganesha/nfsd/ExportMgr  org.ganesha.nfsd.exportmgr.UpdateExport string:${conf_file} string:"EXPORT(Export_Id = ${export_id})"
+
+	sleep 15
+	echo "++++++++Export Data Updated+++++++++"
+fi
+
+if [ "$1" = "server_stage2" ]
+then
+	echo "======= SERVER STAGE 2 =========="
+
+	conf_file="/etc/ganesha/exports/export."${GLUSTER_VOLUME}".conf"
+
+	#Parsing export id from volume export conf file
+	export_id=$(grep 'Export_Id' ${conf_file} | sed 's/^[[:space:]]*Export_Id.*=[[:space:]]*\([0-9]*\).*/\1/')
+
+	sed -i '22s/.*/\t\tAccess_type = "RW";/' ${conf_file}
```

Comment:
These should search-and-replace for the correct lines, I think, rather than changing a specific line by number.